examples/graph/app.js

```diff
@@ -0,0 +1,393 @@
+    window.addEventListener('resize', this._resize);
+    this._resize();
+    this._animate();
+  }
+
+  shouldComponentUpdate() {
+    // Prevent component from updating while animation is running,
+    // to ensure layout updating and rendering are synchronized.
+    return !this._animationFrame;
+  }
+
+  componentDidUpdate() {
+    /* eslint-disable react/no-did-update-set-state */
+    // lastDragged state lasts only one frame
+    const {lastDragged} = this.state;
+    if (lastDragged) {
+      this.setState({
+        lastDragged: null
+      });
+    }
+    /* eslint-enable react/no-did-update-set-state */
+  }
+
+  componentWillUnmount() {
+    this._stopAnimation();
+  }
+
+  _animate() {
+    this.forceUpdate();
+    if (typeof window !== 'undefined') {
+      this._animationFrame = window.requestAnimationFrame(this._animate);
+    }
+  }
+
+  _stopAnimation() {
+    window.cancelAnimationFrame(this._animationFrame);
+    this._animationFrame = null;
+  }
+
+  _resize() {
+    this._onChangeViewport({
+      width: window.innerWidth,
+      height: window.innerHeight
+    });
+  }
+
+  _onChangeViewport(viewport) {
+    this.setState({
+      viewport: {...this.state.viewport, ...viewport}
+    });
+  }
+
+  _onHover(el) {
+    if (el) {
+      this.setState({hovered: el});
+    }
+  }
+
+  _onClick(el) {
+    if (el) {
+      this.setState({clicked: el});
+    } else {
+      const {clicked} = this.state;
+      if (clicked) {
+        this.setState({clicked: null});
+      }
+    }
+  }
+
+  _onDragMove(el) {
+    if (el) {
+      const {viewport} = this.state;
+      const {width, height} = viewport;
+      const x = el.x - width / 2;
+      const y = el.y - height / 2;
+      this.setState({
+        clicked: el,
+        dragging: {
+          node: el.object,
+          x,
+          y
+        }
+      });
+    }
+  }
+
+  _onDragEnd(el) {
+    if (el) {
+      this.setState({
+        dragging: null,
+        lastDragged: {
+          node: el.object
+        }
+      });
+    }
+  }
+
+  //
+  // layout (d3-force) accessors
+  //
+  _linkDistance(link, i) {
+    return 20;
+  }
+
+  _linkStrength(link, i) {
+    if (link.sourceCount || link.targetCount) {
+      return 1 / Math.min(link.sourceCount, link.targetCount);
+    }
+    return 0.5;
+  }
+
+  _nBodyStrength(node, i) {
+    if (node.size) {
+      return -Math.pow(node.size, 1.5) * 3;
+    }
+    return -60;
+  }
+
+  //
+  // rendering (deck.gl) accessors
+  //
+  _getNodeColor(node) {
+    return [18, 147, 154, 255];
+  }
+
+  _getNodeSize(node) {
+    return node.size || 8;
+  }
+
+  // map node type to icon in texture atlas
+  _getNodeIcon(node) {
+    switch (node.type) {
+    case 0:
+      return 'burger';
+    case 1:
+      return 'fries';
+    case 2:
+      return 'soda';
+    case 3:
+    case 4:
+      return 'pie';
+    default:
+      return null;
+    }
+  }
+
+  //
+  // rendering
+  //
+  _renderInteractionLayer(viewport, hovered, clicked) {
```

Comment:
it seems the interactionLayer is an SVG layer used for highlighting the selected & related elements.
we might need some descriptions above for explanation. 

also, (not in this PR), while I agree that using SVG for highlighting should be performant enough given we are only highlighting a limited number of connected/related elements for the current graph use case, we loose the flexibility to interleave them (Interaction layer will also on top) for 3D graphs. let's discuss this later.